### DISCUSSION TITLE
Include customizations in snake info for /move POST

### DISCUSSION THREAD

**Discussion by BoldBigflank** [BattlesnakeOfficial/feedback#129] [Feature Requests]
I am working on making my neopixel light strip show the colors of snakes while my server is playing games with them. Turns out I'm not given enough information to do that. Eventually I'd like to make an LED matrix to show the board as it is going.

I understand this is more information that could bloat the POST requests. Perhaps the customization info could only be in the START post.

**Comment by BoldBigflank** [comment]
I'd also love to have access to the server url of the public snakes in the game.

**Reply by xtagon** [reply]
> I'd also love to have access to the server url of the public snakes in the game.

URLs I don't agree with. Especially since there's no real authentication in place between Battlesnake and individual snake URLs.

But I agree customization/snake info is a good idea!

**Reply by fables-tales** [reply]
yeah I strongly agree we shouldn't give access to those URLs, but I would be down with the customisations being in the payload

**Reply by bvanvugt** [reply]
@xtagon and @penelopezone already stated this, but just to respond officially -- we won't ever expose Battlesnake server URLs to other players/developers 👍

**Comment by jlafayette** [comment]
What about using this API? It seems to be geared more for displaying a game and includes the snake customizations:
https://engine.battlesnake.com/games/48fcc2ea-9642-4b64-b5fb-e14604dd6625/frames?offset=0&limit=1
(with offset=0 and limit=1 it will return just the first turn)

**Reply by BoldBigflank** [reply]
I'm trying this out and works in the primary use case, watching tournament games roll in. It doesn't work if I'm using a local engine or Mojave to run the games.

**Reply by fables-tales** [reply]
I think technically this API isn't officially documented anywhere, and so isn't something we're supposed to rely on?

**Reply by robbles** [reply]
It's not, but feel free to use these public endpoints for low traffic projects like this! There's no guarantees they won't change though :)

You might find this endpoint is a bit more straightforward to use for this:
https://engine.battlesnake.com/games/48fcc2ea-9642-4b64-b5fb-e14604dd6625

**Comment by bvanvugt** [comment]
We discussed this internally - we think including `color`, `head`, and `tail` on snake objects is a good call, and not enough bloat to really worry about. We'll target rolling this out AFTER Fall League 2021, which concludes on Nov 20th.

**Reply by InterestingBrainPoops** [reply]
A leadup announcement with the finalized details / changes would be nice, so that snakes that have a strict serialization scheme can adapt beforehand! But I'm really exited for a v2 api!

**Comment by aurorawalker** [comment]
Whoops! This change was rolled out just at the very end of November 2021, and has been live for a while now. I think in all the excitement of wrapping fall league, we lost track - apologies for that!

Documentation has now been updated (https://docs.battlesnake.com/references/api#battlesnake), and will close this feedback as complete! Thanks.
